kamelets/salesforce-update-sink.kamelet.yaml

```diff
@@ -0,0 +1,111 @@
+    camel.apache.org/provider: "Apache Software Foundation"
+    camel.apache.org/kamelet.group: "Salesforce"
+  labels:
+    camel.apache.org/kamelet.type: "sink"
+spec:
+  definition:
+    title: "Salesforce Update Sink"
+    description: |-
+      Updates an object in  Salesforce. The body received must contain a
+      JSON key-value pair for each property to update or a valid Salesforce
+      object. If using key-value pair, sObjectName and sObjectId must be
+      provided as parameters.
+
+      Example of key-value pair: { "Phone": "1234567890", "Name": "Antonia" }
+    required:
+      - clientId
+      - clientSecret
+      - userName
+      - password
+    type: object
+    properties:
+      sObjectName:
+        title: Object Name
+        description: Type of the object
+        type: string
+        example: Contact
+      sObjectId:
```

Comment:
Is it expected that the user should provide the sObjectId as part of the properties ? 
it seems that this could be very limited as what if you want to update another object ?

Reply:
If they provide the full salesforce object, then that contains the sObjectId.

If they don't provide the full salesforce object, then it must be defined outside the body. That's why sObjectId is not required.

Reply:
> That's why sObjectId is not required.

That's partially true as it is not marked as required parameter but it is not optional in the route so the route is likely failing in case the id is not set.

I would also argue that we should be a little opinionated when making a kamelet as if the end result is about wrapping an endpoint, then there's a very little advantage in using a kamelet vs using the component. Having a non clear interface also make it a little bit more complex to use a Kamelet from tooling POV (Kaoto, Karavan, OpenShifr Dev Console, etc).

That said, I'm fine to keep `sObjectName` and `sObjectId` if needed but let's try to have a concrete use case for when setting the `sObjectId` as part of a kamelet configuration would be required.

Reply:
> That's partially true as it is not marked as required parameter but it is not optional in the route so the route is likely failing in case the id is not set. 

Oh, you are right.

I think using salesforce objects is a more complex usecase than just sending the key-value pairs for attributes changing.

But maybe for real users that use salesforce I am saying a heresy :D

Reply:
I made them truly optional, I don't feel I know salesforce enough to be opinionated :)

Reply:
> I think using salesforce objects is a more complex usecase than just sending the key-value pairs for attributes changing.
> 
> But maybe for real users that use salesforce I am saying a heresy :D

I don't know either and that's why I personal would start with something opinionated and I would iterate over it once we have clarity. Adding an optional field is much easier an safe than removing one as that would break the contract.

Reply:
Then I will work only with key-value pairs and let's see if someone complains :)

Reply:
Pushed.